### PR TITLE
hid: Write to all layouts, implement circular buffers, set up controller metadata.

### PR DESCRIPTION
**Pull request by shinyquagsire23**
Currently all layouts are set up to be written as handheld layouts, but on the actual console these layouts will adjust for controller orientation (ie a single Joy-Con has the face button and stick rotation corrected from the hid service itself for certain layouts, and left alone for others).

A record of the past 17 samples is kept for games to use as well using a circular buffer system with timestamps in number of samples/updates (I believe this may also allow for dropped packets to be picked up on for wireless?)

Additionally the main controller is set up as two neon blue/red Joy-Con docked into a console and charging. Maybe color could be configurable/fetched from actual Joy-Con later?

### CODE REVIEW

File: src/core/hle/service/hid/hid.cpp [hunk 1]
```diff
@@ -65,44 +65,71 @@ class IAppletResource final : public ServiceFramework<IAppletResource> {
...
-        state.sl.Assign(buttons[SL - BUTTON_HID_BEGIN]->GetStatus());
-        state.sr.Assign(buttons[SR - BUTTON_HID_BEGIN]->GetStatus());
-
-        // TODO(shinyquagsire23): Analog stick vals
-
-        // TODO(shinyquagsire23): Update pad info proper, (circular buffers, timestamps, layouts)
+        // Set up controllers as neon red+blue Joy-Con attached to console
+        ControllerHeader& controllerHeader = mem->controllers[Controller_Handheld].header;
+        controllerHeader.type = ControllerType_Handheld | ControllerType_JoyconPair;
+        controllerHeader.singleColorsDescriptor = ColorDesc_ColorsNonexistent;
```
Comment (bunnei) [inline src/core/hle/service/hid/hid.cpp]: Sorry, I missed this in the last PR... but would you mind doing a little bit of naming cleanup? (can be a separate PR if you'd like). See https://github.com/yuzu-emu/yuzu/blob/master/CONTRIBUTING.md (same guidelines as Citra). Member variables and local variables should be `lowercase_with_underscores`, `notCamelCase`.

File: src/core/hle/service/hid/hid.cpp [hunk 2]
```diff
@@ -65,44 +65,71 @@ class IAppletResource final : public ServiceFramework<IAppletResource> {
...
+        controllerHeader.rightColorBody = 0xFF3C28;
```
Comment (Subv) [inline src/core/hle/service/hid/hid.cpp]: Can these be named constants?

Comment (bunnei) [reply src/core/hle/service/hid/hid.cpp]: +1

Comment (shinyquagsire23) [reply src/core/hle/service/hid/hid.cpp]: Sure, though note that these values aren't coded into the OS or anything, the hid service just reads them from Joy-Con SPI flash. Might end up moving to ini settings or something later on idk.

**Comment by bunnei** [comment]
ty lgtm 👍
